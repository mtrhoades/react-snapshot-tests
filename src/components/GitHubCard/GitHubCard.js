import React from 'react';
import Card from 'react-bootstrap/Card';
import Button from 'react-bootstrap/Button';
import GithubImg from '/home/matthew/Desktop/coding/UNLV/Full_Course/submission_activities/snapshot-tests/src/images/github.png';


export default function GitHubCard() {
  return (
    <div style={{display: 'flex', justifyContent: 'center', fontFamily: 'sans-serif'}}>
        <Card style={{ width: '18rem' }}>
            <Card.Img variant="top" src={GithubImg} style={{width: '200px'}} />
            <Card.Body>
                <br></br>
                <Card.Title style={{fontSize: '20px'}}>Matthew Rhoades</Card.Title>
                <Card.Text>
                Github Profile
                </Card.Text>
                <br></br>
                <Button style={{textDecoration: 'none', backgroundColor: 'black', borderRadius: '30px', padding: '10px', color: 'white'}} href="https://github.com/" variant="primary">Go to github</Button>
            </Card.Body>
        </Card>
    </div>
  )
}
