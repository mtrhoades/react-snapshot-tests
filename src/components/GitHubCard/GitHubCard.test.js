import GitHubCard from "./GitHubCard";
import renderer from 'react-test-renderer'

test('renders a snapshot', () => {
    const tree = renderer.create(<GitHubCard />).toJSON(); 
    console.log(tree);
    expect(tree).toMatchSnapshot(); 
})