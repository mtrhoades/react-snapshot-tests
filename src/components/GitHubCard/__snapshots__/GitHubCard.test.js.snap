// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "fontFamily": "sans-serif",
      "justifyContent": "center",
    }
  }
>
  <div
    className="card"
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <img
      className="card-img-top"
      src="github.png"
      style={
        Object {
          "width": "200px",
        }
      }
    />
    <div
      className="card-body"
    >
      <br />
      <div
        className="card-title h5"
        style={
          Object {
            "fontSize": "20px",
          }
        }
      >
        Matthew Rhoades
      </div>
      <p
        className="card-text"
      >
        Github Profile
      </p>
      <br />
      <a
        className="btn btn-primary"
        disabled={false}
        href="https://github.com/"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        style={
          Object {
            "backgroundColor": "black",
            "borderRadius": "30px",
            "color": "white",
            "padding": "10px",
            "textDecoration": "none",
          }
        }
        tabIndex={0}
      >
        Go to github
      </a>
    </div>
  </div>
</div>
`;
